src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/products/ProductSyncWithPricesIT.java

```diff
@@ -0,0 +1,201 @@
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.products.Price;
+import io.sphere.sdk.products.PriceDraft;
+import io.sphere.sdk.products.PriceDraftBuilder;
+import io.sphere.sdk.products.PriceTier;
+import io.sphere.sdk.products.PriceTierBuilder;
+import io.sphere.sdk.products.ProductDraft;
+import io.sphere.sdk.products.ProductDraftBuilder;
+import io.sphere.sdk.products.ProductProjection;
+import io.sphere.sdk.products.ProductProjectionType;
+import io.sphere.sdk.products.ProductVariantDraft;
+import io.sphere.sdk.products.ProductVariantDraftBuilder;
+import io.sphere.sdk.products.commands.ProductCreateCommand;
+import io.sphere.sdk.products.queries.ProductProjectionByKeyGet;
+import io.sphere.sdk.products.queries.ProductProjectionQuery;
+import io.sphere.sdk.producttypes.ProductType;
+import io.sphere.sdk.producttypes.ProductTypeDraft;
+import io.sphere.sdk.producttypes.ProductTypeDraftBuilder;
+import io.sphere.sdk.producttypes.commands.ProductTypeCreateCommand;
+import io.sphere.sdk.utils.MoneyImpl;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ProductSyncWithPricesIT {
+
+  private static final String RESOURCE_KEY = "foo";
+  private static ProductProjectionQuery productQuery;
+
+  private ProductSync productSync;
+  private List<String> errorCallBackMessages;
+  private List<String> warningCallBackMessages;
+  private List<Throwable> errorCallBackExceptions;
+  private ReferenceIdToKeyCache referenceIdToKeyCache;
+
+  @BeforeAll
+  static void setupProjectData() {
+    deleteProductSyncTestData(CTP_TARGET_CLIENT);
+    deleteProductSyncTestData(CTP_SOURCE_CLIENT);
+  }
+
+  @BeforeEach
+  void setup() {
+    errorCallBackMessages = new ArrayList<>();
+    errorCallBackExceptions = new ArrayList<>();
+    warningCallBackMessages = new ArrayList<>();
+
+    final ProductSyncOptions syncOptions =
+        ProductSyncOptionsBuilder.of(CTP_TARGET_CLIENT)
+            .errorCallback(
+                (exception, oldResource, newResource, updateActions) -> {
+                  errorCallBackMessages.add(exception.getMessage());
+                  errorCallBackExceptions.add(exception.getCause());
+                })
+            .warningCallback(
+                (exception, oldResource, newResource) ->
+                    warningCallBackMessages.add(exception.getMessage()))
+            .build();
+    productSync = new ProductSync(syncOptions);
+    referenceIdToKeyCache = new CaffeineReferenceIdToKeyCacheImpl();
+  }
+
+  @AfterEach
+  void tearDownSuite() {
+    deleteProductSyncTestData(CTP_TARGET_CLIENT);
+    deleteProductSyncTestData(CTP_SOURCE_CLIENT);
+  }
+
+  @Test
+  void sync_withNewProductWithPriceTiers_shouldCreateProduct() {
+
+    createAndFetchProductType(CTP_TARGET_CLIENT);
```

Comment:
it seems nothing is fetched here. If it is just for creation of product type in target project, can we put it into `setup()` method?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/734/commits/100750cae39b877f1e05d9909adbb66cf51c3789

It is better to keep this method here, Since I am using common method(prepareDataWithPriceTier) to create productType in both Source and Target clients.